### PR TITLE
added search persistence when moving from bookmark to home

### LINKED ISSUE #224: Search results not persistent when user switches between bookmark and search section
**Describe the bug**

We can easily imagine a user switching between the search and the bookmarks section in a single session. But during this switching, the search results are not persistent. This can become frustrating for regular users as they would have to search again and again or may have to adapt to not switch between the sections. Both of these alternatives do not promise a good User Experience.

ezgif com-video-to-gif (3)

Fixing this would also prevent some unnecessary requests to the API server.

**Additional context**
This happens when the option of storing search results across user sessions is toggled on. Therefore, this inconsistent behavior of the extension is another reason to fix this.

**To Reproduce**
Steps to reproduce the behavior (make sure to uncheck the :
1. Search for something 
2. Switch to the bookmarks section
3. Switch back to the search section.

### PR DESCRIPTION
**Pull request by techievivek**
Fixes #224 

**Description**
Added search persistency when moving from bookmark context to home.
ezgif com-video-to-gif(3)

**Other information**

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by techievivek** [comment]
@makkoncept The issue was when we clicked on the home button, it called the function `loadStoredSearch`, which basically first checks whether the other setting `enableSearchStorage` is checked or not, so when moving from bookmark to home, it creates a problem.
Also, there was issue when someone searches a `query`, so I had to remove the condition and store the result no matter what.
And created 2 different functions, one when the home button gets clicked and the other when the extension is open.
Please verify it.

**Comment by makkoncept** [comment]
@techievivek Thanks for the PR! I will check it out soon.

File: src/popup/popup.js [hunk 1]
```diff
@@ -331,6 +331,24 @@ function setEnableSearchStorageOptionVariable(enableSearchStorage) {
...
+      addThumbnailsToDOM(pageData);
+      pageNo = Number(pageNo) + 1;
+    }
+    elements.clearSearchButton[0].classList.remove('display-none');
+  } else {
+    elements.clearSearchButton[0].classList.add('display-none');
+  }
+}
+
+async function loadStoredSearchOnLoad() {
```
Comment (makkoncept) [inline src/popup/popup.js:351]: `loadStoredSearchOnInit()`  sounds like a better name for this function.

**Review by makkoncept — CHANGES_REQUESTED** [review]
This works fine but I have found a very subtle bug. When the user searches for a query that returns an empty result then the localStorage is not updated as the result array is empty. This means that after switching from the bookmarks section, the last correct result will be loaded. That's good. But, below the loaded images in this particular case, the "No image found" message is still present. Please refer to the gif below:


ezgif com-optimize

**Comment by techievivek** [comment]
@makkoncept Thank you for pointing out the bug, I will fix the bug soon and update the PR.

**Comment by techievivek** [comment]
> This works fine but I have found a very subtle bug. When the user searches for a query that returns an empty result then the localStorage is not updated as the result array is empty. This means that after switching from the bookmarks section, the last correct result will be loaded. That's good. But, below the loaded images in this particular case, the "No image found" message is still present. Please refer to the gif below:
> 
> ezgif com-optimize

The issue was if an invalid input was searched, the localStorage wasn't cleared, and thus moving from Bookmark to Home left with the same result, so once the input has been validated along with other parameters, I clear the localStorage. Please have a look at updated PR.

**Comment by makkoncept** [comment]
After switching from the bookmarks section to the search section, instead of the "No images found" message, the default search section message can be shown.
ezgif com-video-to-gif (5)

**Comment by techievivek** [comment]
@makkoncept I will go through it and update the PR. Thank you

**Comment by techievivek** [comment]
@makkoncept Please have a look at the Updated PR, I have added the required changes.
ezgif com-video-to-gif(4)

**Review by makkoncept — APPROVED** [review]

**Pull request merged**